adr/adr.000.readme.md

```diff
@@ -0,0 +1,86 @@
+# What's an ADR?
```

Comment:
How I see it is:
1.  We discuss together about the approach
2. Someone take the lead and we trust them
3. Once we're 100% of the way we'll move forward we can just ask ChatGPT to generate one that should be 90% okay so it's easier for anyone to go back to if they have questions. Otherwise I feel we can get lost with all the Slack messages and Looms.

For example for me for styling and design system I'm not ready at all. Need to confirm stuff with Andrew and do some more testing. I'm doing progress but I'm not 100% confident I have THE WAY yet so won't do ADR

Reply:
> I never used ADR but down for it! But I think since we're trying to fix lots of things fast for now I wouldn't create an ADR for each thing. And maybe once we're super happy with where we're at we can start making those?

@thierryskoda I think that's a reasonable assessment. ADRs are for standards we're laying forth in the codebase, they are not for every single change we want to make. They should be somewhat "significant", such as, "This is how I propose we organize our files" or "This is how I propose we architect the features in our application" or "This is how we are going to design our component library", etc.

Reply:
> Completely agree with this, there's too many things that we want to change right now to make a full ADR worth it, we need a general understanding/signoff on things, but should feel comfortable setting the patterns. If we feel the pattern starts slipping or needs to be reimagined in the future I think ADRs will be more helpful

@alexrisch I'm gonna push back on this for sure. 

**we need a general understanding/signoff on things** - ADRs are a simple way to do this. I'm happy to do some other strategy if you have one, but unless we have something written / documented somewhere, it isn't fair to provide feedback in a PR that doesn't follow that guideline. Not saying we can agree on everything, but these proposals are certainly to kick off discussions and document tradeoffs and decisions over time. 

Patterns won't just set, we need to be very intentional about them. 

See my [comment above to Theirry](https://github.com/ephemeraHQ/converse-app/pull/1005#discussion_r1803798753) describing when we should use them and not and perhaps that will quell any doubts.

Reply:
Yes I agree that they could be useful.

How I see it is:
1. Create small ADR
2. Have EVERYONE put their questions/thoughts
3. Jump on a call to discuss
4. Finalize the ADR

I think discussion within a PR can be a mess but if the ADR owner can aggregate all the thoughts and questions in one place and one the call we go through them I think it's great! (IMO)

Reply:
Will bring this up in standup! 

That sounds like a great process to me